### PR TITLE
JNG-4327 Add commons functions to ne able to do autoregistering templ…

### PR DESCRIPTION
**Pull request by robertcsakany**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4327" title="JNG-4327" target="_blank">StoryJNG-4327</a>  Auto scan for TemplateHelpers from classpath in generator-maven
  </td></table>
  <br />
 

…ate modules

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+        <maven.build.timestamp.format>yyyyMMddHHmmss</maven.build.timestamp.format>
+        <src.dir>src/main</src.dir>
+
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
+        <surefire-version>3.0.0-M5</surefire-version>
+        <slf4j-version>1.7.36</slf4j-version>
+
+        <!-- Model versions -->
+        <judo-meta-psm-version>1.2.4.20221208_130618_fd953c86_develop</judo-meta-psm-version>
```
Comment (noherczeg) [inline pom.xml]: meta-psm? So we can't use this for ui?

**Review by noherczeg — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (robertcsakany) [reply pom.xml]: Oh. Some leftover :)

**Review by noherczeg — APPROVED** [review]

**Review by gaborflorian — APPROVED** [review]

**Review by fmagnucz — APPROVED** [review]

File: src/main/java/hu/blackbelt/judo/generator/commons/ChainedURLTemplateLoader.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+            }
+            URL scriptUrl = new URI((root + location_rel)).normalize().toURL(); //root.resolve(location).toURL();
+                try (InputStream is = scriptUrl.openStream()) {
+                    if (is != null && is.available() > 0) {
+                        return scriptUrl;
+                    }
+                } catch (Exception e) {
+                }
+        } catch (Exception e) {
+        }
```
Comment (bencegelei) [inline src/main/java/hu/blackbelt/judo/generator/commons/ChainedURLTemplateLoader.java:210]: hm..

**Review by bencegelei — APPROVED** [review]

File: src/main/java/hu/blackbelt/judo/generator/commons/ChainedURLTemplateLoader.java (hunk 2 above)
Comment (fmagnucz) [reply src/main/java/hu/blackbelt/judo/generator/commons/ChainedURLTemplateLoader.java:210]: :-)

**Pull request merged**